### DISCUSSION TITLE
Firebase messaging : aps-environment entitlement

### DISCUSSION THREAD

**Discussion by JeremyBradshaw7** [invertase/react-native-firebase#6450] [Q&A] [answered]
We're adding Firebase messaging to our app. At this stage of the documentation:
https://rnfirebase.io/messaging/usage/ios-setup#enable-background-modes
After carrying out the steps we've noticed the following entitlement added to ./ios/ccf/ccf.entitlements:
![image](https://user-images.githubusercontent.com/36155348/183065163-dafe2e46-63b7-42d6-9f00-8c9fbfeff92a.png)

When merging with production, does this new entitlement have to be retained and should the value be updated from `development` to `production` ? I would assume so but some confirmation would be useful. I cannot find any instructions about this in the docs. 

NB. we use automatic provisioning by specifying `-allowProvisioningUpdates` in `xcodebuild`.

**Comment by mikehardy** [comment]
Sorry for the delay - I am honestly not sure - the entitlements file is generated I think? I have not changed that personally - I imagine you have already deployed to production at least once and "it just worked", right? That's my experience at any rate

**Comment by JeremyBradshaw7** [comment, marked as answer]
That's right, I changed it to `production` and had no issues with that on live.

**Comment by muammadibal** [comment]
> We're adding Firebase messaging to our app. At this stage of the documentation: https://rnfirebase.io/messaging/usage/ios-setup#enable-background-modes After carrying out the steps we've noticed the following entitlement added to ./ios/ccf/ccf.entitlements: ![image](https://user-images.githubusercontent.com/36155348/183065163-dafe2e46-63b7-42d6-9f00-8c9fbfeff92a.png)
> 
> When merging with production, does this new entitlement have to be retained and should the value be updated from `development` to `production` ? I would assume so but some confirmation would be useful. I cannot find any instructions about this in the docs.
> 
> NB. we use automatic provisioning by specifying `-allowProvisioningUpdates` in `xcodebuild`.

what does it ccf? just create folder ccf with ccf.entitlements files with this?
```
<key>aps-environment</key>
<string>production</string>
```
